src/Marten/Linq/QueryHandlers/QueryHandlerFactory.cs

```diff
@@ -157,7 +162,7 @@ private IQueryHandler<T> tryFindSingleQuery<T>(QueryModel model, IIncludeJoin[]
 
             var setters = findSetters(queryType, expression);
 
-            var model = MartenQueryParser.Flyweight.GetParsedQuery(invocation);
+            var model = MartenQueryParser.ForCachedQuery.GetParsedQuery(invocation);
```

Comment:
That static public field is accessed in other places than just for the cached queries, and isn't really specific to cached queries anyway. Could you rename it back to what it was, or something that makes more sense to you if you want? It is just a GoF "flyweight", but feel free to call it "Instance" or something if you prefer